lib/pxutil.js

```diff
@@ -162,6 +76,22 @@ function filterSensitiveHeaders(headers) {
     }
 }
 
+function generateProxyHeaders(headers, ip) {
+    try {
+        const pxConfig = require('./pxconfig').conf;
+        let filteredHeaders = filterSensitiveHeaders(headers);
+        let xffHeaderMatch = Object.keys(filteredHeaders).filter(item => {return item.toLowerCase() === pxConfig.FORWARDED_FOR_HEADER});
```

Comment:
You can use a pattern like `Object.keys(filteredHeaders).find(item => item.toLowerCase() === 'x-forwarded-for')` to find the key for the requested header